### DISCUSSION TITLE
module keeps rebooting

### DISCUSSION THREAD

**Discussion by SpatialJ** [jomjol/AI-on-the-edge-device#1041] [Q&A] [answered]
Hi,
im trying to get the project running on this module:
https://arduino-projekte.info/produkt/esp32-cam-v2-integriertem-ch340-mit-ov2640-kamera-modul/
I flashed the firmware and tried to different, brand new sd-cards so far, but the device just keeps rebooting.
Here are the logs from serial monitor, does anybody has an idea whats wrong in my setup?

Thanks,
Johannes

20:28:21.708 -> Rebooting...
20:28:21.708 -> ets Jun  8 2016 00:22:57
20:28:21.708 -> 
20:28:21.708 -> rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
20:28:21.708 -> configsip: 0, SPIWP:0xee
20:28:21.708 -> clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
20:28:21.708 -> mode:DIO, clock div:2
20:28:21.708 -> load:0x3fff0030,len:7580
20:28:21.708 -> ho 0 tail 12 room 4
20:28:21.741 -> load:0x40078000,len:15360
20:28:21.741 -> load:0x40080400,len:3744
20:28:21.741 -> entry 0x4008067c
20:28:21.741 -> [0;32mI (29) boot: ESP-IDF 4.3.2 2nd stage bootloader[0m
20:28:21.741 -> [0;32mI (29) boot: compile time 21:14:26[0m
20:28:21.741 -> [0;32mI (29) boot: chip revision: 1[0m
20:28:21.741 -> [0;32mI (32) boot_comm: chip revision: 1, min. bootloader chip revision: 0[0m
20:28:21.742 -> [0;32mI (39) boot.esp32: SPI Speed      : 40MHz[0m
20:28:21.742 -> [0;32mI (43) boot.esp32: SPI Mode       : DIO[0m
20:28:21.773 -> [0;32mI (48) boot.esp32: SPI Flash Size : 4MB[0m
20:28:21.773 -> [0;32mI (52) boot: Enabling RNG early entropy source...[0m
20:28:21.773 -> [0;32mI (58) boot: Partition Table:[0m
20:28:21.773 -> [0;32mI (61) boot: ## Label            Usage          Type ST Offset   Length[0m
20:28:21.773 -> [0;32mI (69) boot:  0 nvs              WiFi data        01 02 00009000 00004000[0m
20:28:21.773 -> [0;32mI (76) boot:  1 otadata          OTA data         01 00 0000d000 00002000[0m
20:28:21.806 -> [0;32mI (84) boot:  2 phy_init         RF data          01 01 0000f000 00001000[0m
20:28:21.806 -> [0;32mI (91) boot:  3 ota_0            OTA app          00 10 00010000 001db000[0m
20:28:21.806 -> [0;32mI (99) boot:  4 ota_1            OTA app          00 11 001f0000 001db000[0m
20:28:21.806 -> [0;32mI (106) boot: End of partition table[0m
20:28:21.850 -> [0;32mI (110) boot_comm: chip revision: 1, min. application chip revision: 0[0m
20:28:21.850 -> [0;32mI (117) esp_image: segment 0: paddr=00010020 vaddr=3f400020 size=4b7b8h (309176) map[0m
20:28:21.975 -> [0;32mI (244) esp_image: segment 1: paddr=0005b7e0 vaddr=3ffb0000 size=04838h ( 18488) load[0m
20:28:22.491 -> [0;32mI (775) esp_image: segment 3: paddr=001ad42c vaddr=3ffb4838 size=001a0h (   416) load[0m
20:28:22.491 -> [0;32mI (775) esp_image: segment 4: paddr=001ad5d4 vaddr=40080000 size=1bc7ch (113788) load[0m
20:28:22.523 -> [0;32mI (830) esp_image: segment 5: paddr=001c9258 vaddr=50000000 size=00010h (    16) load[0m
20:28:22.555 -> [0;32mI (844) boot: Loaded app from partition at offset 0x10000[0m
20:28:22.555 -> [0;32mI (844) boot: Disabling RNG early entropy source...[0m
20:28:22.555 -> [0;32mI (856) psram: This chip is ESP32-D0WD[0m
20:28:22.555 -> [0;32mI (856) spiram: Found 64MBit SPI RAM device[0m
20:28:22.555 -> [0;32mI (856) spiram: SPI RAM mode: flash 40m sram 40m[0m
20:28:22.599 -> [0;32mI (861) spiram: PSRAM initialized, cache is in low/high (2-core) mode.[0m
20:28:22.599 -> [0;32mI (868) cpu_start: Pro cpu up.[0m
20:28:22.599 -> [0;32mI (872) cpu_start: Starting app cpu, entry point is 0x40082c58[0m
20:28:22.600 -> [0;32mI (864) cpu_start: App cpu up.[0m
20:28:23.475 -> [0;32mI (1766) spiram: SPI SRAM memory test OK[0m
20:28:23.475 -> [0;32mI (1774) cpu_start: Pro cpu start user code[0m
20:28:23.475 -> [0;32mI (1774) cpu_start: cpu freq: 160000000[0m
20:28:23.475 -> [0;32mI (1774) cpu_start: Application information:[0m
20:28:23.508 -> [0;32mI (1777) cpu_start: Project name:     esp32cam-server-only[0m
20:28:23.508 -> [0;32mI (1783) cpu_start: App version:      0.0.9.3[0m
20:28:23.508 -> [0;32mI (1788) cpu_start: Compile time:     Sep 17 2022 21:13:04[0m
20:28:23.508 -> [0;32mI (1795) cpu_start: ELF file SHA256:  07220ec4513ba756...[0m
20:28:23.508 -> [0;32mI (1801) cpu_start: ESP-IDF:          4.3.2[0m
20:28:23.508 -> [0;32mI (1806) heap_init: Initializing. RAM available for dynamic allocation:[0m
20:28:23.542 -> [0;32mI (1813) heap_init: At 3FFAE6E0 len 00001920 (6 KiB): DRAM[0m
20:28:23.542 -> [0;32mI (1819) heap_init: At 3FFC0888 len 0001F778 (125 KiB): DRAM[0m
20:28:23.542 -> [0;32mI (1825) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM[0m
20:28:23.542 -> [0;32mI (1832) heap_init: At 3FFE4350 len 0001BCB0 (111 KiB): D/IRAM[0m
20:28:23.542 -> [0;32mI (1838) heap_init: At 4009BC7C len 00004384 (16 KiB): IRAM[0m
20:28:23.576 -> [0;32mI (1845) spiram: Adding pool of 4096K of external SPI memory to heap allocator[0m
20:28:23.576 -> [0;32mI (1854) spi_flash: detected chip: generic[0m
20:28:23.576 -> [0;32mI (1857) spi_flash: flash io: dio[0m
20:28:23.576 -> [0;32mI (1867) cpu_start: Starting scheduler on PRO CPU.[0m
20:28:23.576 -> [0;32mI (0) cpu_start: Starting scheduler on APP CPU.[0m
mory for DMA/internal allocations[0m
20:28:23.616 -> [0;32mI (1882) gpio: GPIO[32]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0 [0m
20:28:25.580 -> Init Camera
20:28:25.580 -> [0;32mI (3892) gpio: GPIO[25]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:2 [0m
20:28:25.624 -> [0;32mI (3892) cam_hal: cam init ok[0m
20:28:25.624 -> [0;32mI (3892) sccb: pin_sda 26 pin_scl 27[0m
20:28:25.624 -> [0;32mI (3892) gpio: GPIO[32]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0 [0m
20:28:25.703 -> [0;31mE (4012) camera: Camera probe failed with error 0x105(ESP_ERR_NOT_FOUND)[0m
20:28:25.703 -> [0;31mE (4012) server_part_camera: Camera Init Failed[0m
20:28:25.750 -> Internal Flash-LED turn off PWM
20:28:25.750 -> nach init camera: sleep for : 200ms
20:28:27.785 -> [0;32mI (6052) main: Using SDMMC peripheral[0m
20:28:27.817 -> Name: USD00
20:28:27.817 -> Type: SDHC/SDXC
20:28:27.817 -> Speed: 20 MHz
20:28:27.817 -> Size: 15080MB
20:28:27.817 -> [0;32mI (6102) server_ota: Start CheckOTAUpdateCheck ...[0m
20:28:27.817 -> Start CheckOTAUpdateCheck ...
20:28:27.817 -> [0;32mI (6112) server_ota: SHA-256 for the partition table: : 9d4a2809ae90fb20709a8ca3dd3ac3462e1e73286bcd31b689060c91de06fa3a[0m
20:28:27.817 -> [0;32mI (6112) server_ota: SHA-256 for bootloader: : 349af213bfec9b5048e04c3179ddcb542fb9b183c649e296f16fe7f11f5c59e4[0m
20:28:27.853 -> [0;32mI (6122) server_ota: SHA-256 for current firmware: : 1bc0035189be1e62ea8e983c8b67c6c4ac17a90854535610634817b812766253[0m
20:28:27.853 -> CheckOTAUpdate Partition: ESP_OK
20:28:27.887 -> open file /sdcard/wlan.ini in mode r
20:28:27.887 -> file loaded
20:28:27.887 -> 
20:28:27.887 -> WLan: xxxxxxxxxxxxxxxxxxx, xxxxxxxxxxxxxxxxxxxxx

20:28:27.887 -> I (6192) wifi:wifi driver task: 3ffd2b6c, prio:23, stack:6656, core=0
20:28:27.887 -> [0;32mI (6192) system_api: Base MAC address is not set[0m
20:28:27.919 -> [0;32mI (6192) system_api: read default base MAC address from EFUSE[0m
20:28:27.919 -> I (6212) wifi:wifi firmware version: eb52264
20:28:27.919 -> I (6212) wifi:wifi certification version: v7.0
20:28:27.919 -> I (6212) wifi:config NVS flash: enabled
20:28:27.919 -> I (6212) wifi:config nano formating: disabled
20:28:27.919 -> I (6222) wifi:Init data frame dynamic rx buffer num: 16
20:28:27.952 -> I (6222) wifi:Init management frame dynamic rx buffer num: 16
20:28:27.952 -> I (6232) wifi:Init management short buffer num: 32
20:28:27.952 -> I (6232) wifi:Init static tx buffer num: 16
20:28:27.952 -> I (6242) wifi:Init tx cache buffer num: 16
20:28:27.952 -> I (6242) wifi:Init static rx buffer size: 1600
20:28:27.952 -> I (6242) wifi:Init static rx buffer num: 10
20:28:27.952 -> I (6252) wifi:Init dynamic rx buffer num: 16
20:28:27.986 -> [0;32mI (6252) wifi_init: rx ba win: 6[0m
20:28:27.986 -> [0;32mI (6262) wifi_init: tcpip mbox: 32[0m
20:28:27.986 -> [0;32mI (6262) wifi_init: udp mbox: 6[0m
20:28:27.986 -> [0;32mI (6262) wifi_init: tcp mbox: 6[0m
20:28:27.986 -> [0;32mI (6272) wifi_init: tcp tx win: 5744[0m
20:28:27.986 -> [0;32mI (6272) wifi_init: tcp rx win: 5744[0m
20:28:27.986 -> [0;32mI (6282) wifi_init: tcp mss: 1440[0m
20:28:27.986 -> [0;32mI (6282) wifi_init: WiFi IRAM OP enabled[0m
20:28:27.986 -> [0;32mI (6282) wifi_init: WiFi RX IRAM OP enabled[0m
20:28:28.034 -> [0;32mI (6292) phy_init: phy_version 4670,719f9f6,Feb 18 2021,17:07:07[0m
20:28:28.115 -> I (6402) wifi:mode : sta (30:c6:f7:05:29:1c)
20:28:28.115 -> I (6402) wifi:enable tsf
20:28:28.115 -> [0;32mI (6402) BLINK: Blinken - start[0m

20:28:28.115 -> [0;31mE (6412) wifi station: Set Hostname to:wmz[0m
20:28:28.115 -> [0;32mI (6412) wifi station: wifi_init_sta finished.[0m
20:28:28.115 -> I (6412) wifi:new:<6,0>, old:<1,0>, ap:<255,255>, sta:<6,0>, prof:1
20:28:28.158 -> I (6422) wifi:state: init -> auth (b0)
20:28:28.330 -> [0;32mI (6602) BLINK: Blinken - done[0m
20:28:29.144 -> I (7422) wifi:state: auth -> init (200)
20:28:29.144 -> I (7422) wifi:new:<6,0>, old:<6,0>, ap:<255,255>, sta:<6,0>, prof:1
20:28:29.144 -> [0;32mI (7422) wifi station: retry to connect to the AP[0m
20:28:29.144 -> [0;32mI (7422) wifi station: connect to the AP fail[0m
20:28:31.205 -> [0;32mI (9472) wifi station: retry to connect to the AP[0m
20:28:31.205 -> I (9472) wifi:new:<6,0>, old:<6,0>, ap:<255,255>, sta:<6,0>, prof:1
20:28:31.205 -> I (9482) wifi:state: init -> auth (b0)
20:28:31.205 -> [0;32mI (9472) wifi station: connect to the AP fail[0m
20:28:31.205 -> I (9482) wifi:state: auth -> assoc (0)
20:28:31.205 -> I (9492) wifi:state: assoc -> run (10)
20:28:31.248 -> I (9532) wifi:connected with FRITZ!Box 7590 CR, aid = 8, channel 6, BW20, bssid = 2c:3a:fd:2d:cf:84
20:28:31.248 -> I (9532) wifi:security: WPA2-PSK, phy: bgn, rssi: -67
20:28:31.248 -> I (9532) wifi:pm start, type: 1
20:28:31.248 -> 
20:28:31.282 -> W (9562) wifi:<ba-add>idx:0 (ifx:0, 2c:3a:fd:2d:cf:84), tid:6, ssn:3, winSize:64
20:28:31.282 -> I (9572) wifi:AP's beacon interval = 102400 us, DTIM period = 1
20:28:31.878 -> [0;32mI (10172) esp_netif_handlers: sta ip: 192.168.178.57, mask: 255.255.255.0, gw: 192.168.178.1[0m
20:28:31.878 -> [0;32mI (10172) wifi station: got ip:192.168.178.57[0m
20:28:31.878 -> [0;32mI (10172) BLINK: Blinken - start[0m

20:28:31.878 -> [0;32mI (10172) wifi station: connected to ap SSID:xxxxxxxxxxxxxxxxx password:xxxxxxxxxxxxxxx[0m
20:28:31.923 -> main: sleep for : 200ms
20:28:33.878 -> [0;32mI (12192) sntp: Time is not set yet. Connecting to WiFi and getting time over NTP.[0m
20:28:33.923 -> [0;32mI (12192) sntp: Initializing SNTP[0m
20:28:33.923 -> [0;32mI (12192) sntp: Waiting for system time to be set... (1/10)[0m
20:28:35.906 -> [0;32mI (14192) sntp: Waiting for system time to be set... (2/10)[0m
20:28:36.000 -> W (14282) wifi:<ba-add>idx:1 (ifx:0, 2c:3a:fd:2d:cf:84), tid:0, ssn:0, winSize:64
20:28:36.047 -> W (14322) wifi:<ba-add>idx:2 (ifx:0, 2c:3a:fd:2d:cf:84), tid:5, ssn:0, winSize:64
20:28:37.885 -> TimeZone set to CET-1CEST,M3.5.0,M10.5.0/3
20:28:37.919 -> Time zone set to CET-1CEST,M3.5.0,M10.5.0/3
20:28:37.919 -> [0;32mI (16202) sntp: The current date/time in Berlin is: Tue Sep 20 20:28:39 2022[0m
20:28:37.919 -> [0;32mI (16202) sntp: The current date/time in Berlin is: 2022-09-20_20:28[0m
20:28:37.919 -> timeist 20220920-202839
20:28:37.919 -> =============================================================================================
20:28:37.919 -> =================================== Main Started ============================================
20:28:37.953 -> =============================================================================================
20:28:37.953 -> time 20220920-202839
20:28:37.953 -> [0;31mE (16232) main: Failed to initialize camera module. Check that your camera module is working and connected properly.[0m
20:28:37.953 -> Failed to initialize camera module. Check that your camera module is working and connected properly.
20:28:37.993 -> main: sleep for : 2000ms
20:28:39.972 -> starting server
20:28:39.972 -> [0;32mI (18262) server-main: Starting server on port: '80'[0m
20:28:39.972 -> [0;32mI (18262) server-main: Registering URI handlers[0m
20:28:39.972 -> [0;32mI (18262) server_tflite: server_part_camera - Registering URI handlers[0m
20:28:39.972 -> [0;32mI (18262) server_ota: server_ota - Registering URI handlers[0m
20:28:39.972 -> [0;32mI (18272) server_GPIO: start GpioHandler[0m
20:28:40.005 -> [0;32mI (18272) server_GPIO: register GPIO Uri[0m
20:28:40.005 -> [0;32mI (18282) server_GPIO: server_GPIO - Registering URI handlers[0m

20:28:40.005 -> vor reg server main
20:28:40.005 -> vor dotautostart
20:28:40.005 -> task_autodoFlow configMINIMAL_STACK_SIZE: 768
20:28:40.005 -> getESPHeapInfo:  Heap: 4245442 Min Free: 4244642 larg. Block:  4128768 SPI Heap: 4165211 NOT_SPI Heap: 80231 Internal Heap: 80231 Internal Min Heap free: 79943
20:28:40.039 -> task_autodoFlow: start

20:28:40.039 -> getESPHeapInfo:  Heap: 4218218 Min Free: 4217486 larg. Block:  4128768 SPI Heap: 4165211 NOT_SPI Heap: 53007 Internal Heap: 53007 Internal Min Heap free: 52787
20:28:40.039 -> Start tfliteflow.InitFlow(config);
20:28:40.039 -> open file /sdcard/config/config.ini in mode r
20:28:40.039 -> [MakeImage]

20:28:40.039 -> Start ReadParameter ([MakeImage]

20:28:40.039 -> )
20:28:40.039 -> ;LogImageLocation = /log/source

20:28:40.039 -> WaitBeforeTakingPicture = 5

20:28:40.073 -> ;LogfileRetentionInDays = 15

20:28:40.073 -> Brightness = 0

20:28:40.073 -> Contrast = 0

20:28:40.073 -> Saturation = 0

20:28:40.073 -> LEDIntensity = 50

20:28:40.073 -> Set led_intensity to 4095 of 8191
20:28:40.073 -> ImageQuality = 12

20:28:40.073 -> ImageSize = VGA

20:28:40.073 -> FixedExposure = false

20:28:40.073 -> 

20:28:40.073 -> [Alignment]

20:28:40.073 -> Guru Meditation Error: Core  0 panic'ed (LoadProhibited). Exception was unhandled.

20:28:40.073 -> 
20:28:40.073 -> Core  0 register dump:
20:28:40.073 -> PC      : 0x400f08af  PS      : 0x00060130  A0      : 0x8018e11b  A1      : 0x3fff4dc0  
20:28:40.105 -> A2      : 0x00000000  A3      : 0x00000000  A4      : 0x00000000  A5      : 0x00000000  
20:28:40.105 -> A6      : 0x3ffb74cc  A7      : 0x00000001  A8      : 0x00000002  A9      : 0x00000002  
20:28:40.106 -> A10     : 0x00000000  A11     : 0x00000000  A12     : 0x3ffc2c0b  A13     : 0x00000000  
20:28:40.106 -> A14     : 0x00000000  A15     : 0xff000000  SAR     : 0x00000018  EXCCAUSE: 0x0000001c  
20:28:40.139 -> EXCVADDR: 0x0000004c  LBEG    : 0x4008ab70  LEND    : 0x4008ab8c  LCOUNT  : 0xffffffff  
20:28:40.139 -> 
20:28:40.139 -> Backtrace:0x400f08ac:0x3fff4dc0 0x4018e118:0x3fff4de0 0x4018c249:0x3fff4e80 0x400fe785:0x3fff52d0 0x400fea60:0x3fff5310
20:28:40.139 -> 
20:28:40.139 -> 
20:28:40.139 -> ELF file SHA256: 07220ec4513ba756
20:28:40.139 -> 
20:28:40.139 -> Rebooting...`

**Comment by jomjol** [comment]
That looks like a hardware failure:

```
20:28:37.953 -> �[0;31mE (16232) main: Failed to initialize camera module. Check that your camera module is working and connected properly.�[0m
20:28:37.953 -> Failed to initialize camera module. Check that your camera module is working and connected properly.
```

Try to unplug and replug the camera, maybe it is only a connection problem.

**Comment by SpatialJ** [comment]
Thanks for the hint, i'll check.

**Comment by SpatialJ** [comment]
The camera works with this sketch https://randomnerdtutorials.com/esp32-cam-video-streaming-face-recognition-arduino-ide/ as a "CAMERA_MODEL_AI_THINKER", so it shouldnt be a hardware issue.

**Comment by SpatialJ** [comment, marked as answer]
I just reflashed the newest firmware and now the module starts up :-)
Thanks for the help!
